### ISSUE TITLE
VX and VY need to be moved back

### ISSUE THREAD

**Issue by tobiasvl** [tobiasvl/chiposlo#1]
Dream Invaders expects VX at `$002E` and VY at `$002F`.

**Comment by tobiasvl** [comment]
A nice way to do this would be if I could move the stack from `$005F`–`$007F` (to for example `$004F`–`$006F`) and then fit the following somewhere in the setup:

```
LDAA #$39 ; RTS
STAA $7A  ; Store RTS at $007A
```

Then, since VX and VY would've been moved back to `$2E` and `$2F` respectively, these would be BLE/BGT +39, and since that'd be at `$0041` they'd branch to a separate RTS.

But I don't think I have 4 bytes, and it'd be a shame if some other program mangled the stack in its new location.

**Comment by tobiasvl** [comment]
Note that it's actually only VX that I currently need to be at `$000A`, since `$0A` is CLV. Not sure why I moved VY too originally.

**Comment by tobiasvl** [comment]
Even cleaner than the BLE/BGT option would of course be to put VX at `$0039`, so it's an RTS directly. But currently that's in the middle of the V table itself (V9). I doubt other programs access the CHIP-8 variables directly, but Dream Invaders does access the flag variable VF, which is at `$003F`. Consequently, V9 needs to be at `$0039` as well. Sigh.

**Comment by tobiasvl** [comment]
All right, I solved this for the time being.

VX and VY are now back at `$002E` and `$002F` respectively, but `$000A` is now its own VXTMP (meaning DDPAT still needs to remain where I moved it, for the time being). During the opcode fetch, I store the VX value in both VX and VXTMP. To make room for the extra byte this required, I managed to rewrite the EXCALL routine slightly (branching to an `RTS`…).
